In [1]:
import os
for f in os.listdir('../fun/'): exec(open('../fun/'+f).read())
del f

# Load data. Change winner to binary 1/0:
load( '../out/d3-fight-level-standardize-normalize.pkl' )
y[ y == -1 ] = 0
print( X.shape )

(3576, 115)


In [2]:
# train test split.
from sklearn.model_selection import train_test_split
X_train , X_test, y_train, y_test = train_test_split(
    X, y, 
    random_state = 718,
    test_size = 0.25
)

In [3]:
# Fit decision tree.
from sklearn.ensemble import RandomForestRegressor
m = RandomForestRegressor( 
    n_estimators = 400, 
    min_samples_leaf = 0.05, 
    random_state = 841 
)
m.fit( X_train, y_train )

RandomForestRegressor(bootstrap=True, criterion='mse', max_depth=None,
                      max_features='auto', max_leaf_nodes=None,
                      min_impurity_decrease=0.0, min_impurity_split=None,
                      min_samples_leaf=0.05, min_samples_split=2,
                      min_weight_fraction_leaf=0.0, n_estimators=400,
                      n_jobs=None, oob_score=False, random_state=841, verbose=0,
                      warm_start=False)

In [4]:
# accuracy against train data (in-model).
from sklearn.metrics import confusion_matrix, classification_report, f1_score, precision_score, recall_score
print( classification_report( 
    y_train, 
    [ 1 if x > .5 else 0 for x in m.predict(X_train) ] 
))

              precision    recall  f1-score   support

           0       0.69      0.30      0.42       903
           1       0.72      0.93      0.81      1779

    accuracy                           0.72      2682
   macro avg       0.71      0.62      0.62      2682
weighted avg       0.71      0.72      0.68      2682



In [5]:
# accuracy against test data.
from sklearn.metrics import confusion_matrix, classification_report, f1_score, precision_score, recall_score
print( classification_report( 
    y_test, 
    [ 1 if x > .5 else 0 for x in m.predict(X_test) ] 
))

              precision    recall  f1-score   support

           0       0.55      0.22      0.31       301
           1       0.70      0.91      0.79       593

    accuracy                           0.68       894
   macro avg       0.63      0.56      0.55       894
weighted avg       0.65      0.68      0.63       894



f1-score in-model is higher than on test indicating **over-fitting**. 

Test-set precision is still not great at 69% (predicting all-wins give 60%).